go-controller/pkg/config/config.go

```diff
@@ -1149,6 +1173,21 @@ func pathExists(path string) bool {
 	return true
 }
 
+// joinSubnetOverlap returns an err if joinSubnet range contains the subnet
+func overlapsWithJoinSubnet(subnets []*net.IPNet) error {
+	joinSubnet := V4JoinSubnet
+	if IPv6Mode {
+		joinSubnet = V6JoinSubnet
+	}
```

Comment:
better to get ahead of the dual stack curve now; don't look at IPv6Mode, just look at each individual `subnet.IP`.

Actually, you're allowed to ask if an IPv4 CIDR contains an IPv6 IP or vice versa. (It will always be `false` of course, but it doesn't throw an error or anything.) So just check each element of `subnets` against both `V4JoinSubnet` and `V6JoinSubnet`.

Reply:
Will do.